File: Topic0102/008_eucids_algorithm/eucids_algorithm.cpp
#include <iostream>

int gcd(int, int);

int main(){
  int a,  b;
  
  std::cin >> a >> b;

std::cout << a << " and " << b << "\'s GCD is " << gcd(a, b) << std::endl;

  return 0;
}

int gcd(int x, int y){
  for(int i = 0; i < x + y; i++){
    if(x == y)
      break;
    if(x > y)
      x -= y;
    else
      y -= x;
  }
  return x;
}

File: Topic0102/008_eucids_algorithm/eucids_algorithm.js
var x = 100,
  y = 20;

function gcd(a, b) {
  while (a != b) {
    if (a > b) a -= b;
    else b -= a;
  }
  return a;
}

console.log(`GCD of ${x} and ${y} is ${gcd(x, y)}.`);
